### PR TITLE
Unique username

### PR DESCRIPTION
**Pull request by kezzayuno**
- Unique username wasn't working but I just had to do some small changes 
- Make the Toast (suggestion: make it so that the toast is in the middle of the screen, thing is--apparently .setGravity isn't a thing for API 30+, so we're out of luck I think. But we can use snackbar instead. Stretch goal, though. Not a big deal). 
- Save the username immediately instead of passing it around. I called .updateDB() on playerInfo and then remove backstack for the fragment (for profile). That way, when they go "back" to profile, it is the updated profile (not the old one). They doesn't mess up the stack when trying to go back to camera either I don't think (when I tested it, it didn't break anything, and when I think of a stack, it makes sense to me to why it should work).

### CODE REVIEW

File: app/src/main/java/com/bigyoshi/qrhunt/player/FragmentProfile.java [hunk 1]
```diff
@@ -58,62 +50,43 @@
...
+                            @Override
+                            public void handleOnBackPressed() {
+
+                                Intent intent = new Intent(getContext(), MainActivity.class);
+                                Bundle prevNav = new Bundle();
+                                prevNav.putSerializable("previous", lastDestination);
+                                intent.putExtras(prevNav);
+                                startActivity(intent);
+                            }
+                        });
```
Comment (Mattwmaster58) [inline app/src/main/java/com/bigyoshi/qrhunt/player/FragmentProfile.java:71]: honestly hate this style of formatting, i wish there was good java auto formatters out there

but yeah this was my own doing

**Review by Mattwmaster58 — APPROVED** [review]
wow, this pull request seemed to have ballooned in scope! try to keep it focused to one thing eh?

oh wait, that was all me

lgtm

Does it actually save right now?

File: app/src/main/java/com/bigyoshi/qrhunt/player/FragmentProfile.java (hunk 1 above)
Comment (kezzayuno) [reply app/src/main/java/com/bigyoshi/qrhunt/player/FragmentProfile.java:71]: Oh, by the way. I am trying to customize how the error message looks, so it doesn't look so out of place.

Comment (kezzayuno) [reply app/src/main/java/com/bigyoshi/qrhunt/player/FragmentProfile.java:71]: Oh wow this is very annoying. It's a stretch goal now.

**Pull request merged**